src/lib.rs

```diff
@@ -72,30 +72,24 @@ pub fn run_and_output(config: Config) {
 
 const MAX_FIX_ITERATIONS: usize = 10;
 
-#[allow(clippy::too_many_arguments)]
 fn run_per_file<'a>(
-    aggregated_queries: &'a AggregatedQueries<'a>,
-    path: &'a Path,
-    config: &'a Config,
+    file_run_context: FileRunContext<'a>,
     mut on_found_violations: impl FnMut(Vec<ViolationWithContext>),
     mut on_found_pending_fixes: impl FnMut(Vec<PendingFix>, &InstantiatedRule),
-    file_contents: impl Into<RopeOrSlice<'a>>,
-    tree: &'a Tree,
-    query: &'a Arc<Query>,
-    language: SupportedLanguage,
-    instantiated_rules: &'a [InstantiatedRule],
 ) {
-    let file_contents = file_contents.into();
     let mut instantiated_per_file_rules: HashMap<RuleName, Box<dyn RuleInstancePerFile<'a> + 'a>> =
         Default::default();
-    get_matches(language.language(), file_contents, query, Some(tree)).for_each(|query_match| {
+    get_matches(
+        file_run_context.language.language(),
+        file_run_context.file_contents,
+        file_run_context.query,
+        Some(file_run_context.tree),
+    )
+    .for_each(|query_match| {
         run_match(
+            file_run_context,
             query_match,
-            aggregated_queries,
-            language,
-            path,
-            file_contents,
-            config,
+            file_run_context.aggregated_queries,
```

Comment:
Not sure why this is (still) being passed as a separate param when it's available on `file_run_context`